Compiler/src/Compiler.cpp

```diff
@@ -1477,6 +1480,70 @@ struct Compiler
         }
     }
 
+    void compileExprInterpString(AstExprInterpString* expr, uint8_t target, bool targetTemp)
+    {
+        std::string formatString;
+
+        size_t stringsLeft = expr->strings.size;
+
+        for (AstArray<char> const& string : expr->strings)
+        {
+            std::string stringEscaped(string.data);
+
+            for (size_t characterIndex = 0; characterIndex < stringEscaped.size(); ++characterIndex)
+            {
+                if (stringEscaped[characterIndex] == '%')
+                {
+                    stringEscaped.insert(characterIndex, 1, '%');
+                    characterIndex++;
+                }
+            }
+
+            formatString += stringEscaped;
+
+            stringsLeft--;
+
+            if (stringsLeft > 0)
+                formatString += "%*";
+        }
+
+        auto formatStringSize = formatString.size();
+
+        // We can't use formatStringRef.data() directly, because short strings don't have their data
+        // pinned in memory, so when interpFormatStrings grows, these pointers will move and become invalid.
+        std::shared_ptr<char[]> formatStringPtr(new char[formatStringSize]);
+        memcpy(formatStringPtr.get(), formatString.data(), formatStringSize);
+
+        auto formatStringPtrRef = interpFormatStrings.emplace_back(formatStringPtr);
```

Comment:
Shouldn't use auto, also you should be able to simply use formatStringPtr below.